crates/pallet-currency-swap/src/benchmarking.rs

```diff
@@ -0,0 +1,123 @@
+//! The benchmarks for the pallet.
+
+use frame_benchmarking::benchmarks;
+use frame_support::{assert_ok, dispatch::DispatchResult, traits::Currency};
+use frame_system::RawOrigin;
+
+use crate::*;
+
+const WITHDRAW_BALANCE: u32 = 100;
+const DEPOSIT_BALANCE: u32 = 100;
+
+/// The benchmarking extension for the currency swap interface.
+pub trait CurrencySwap<AccountIdFrom, AccountIdTo>:
```

Comment:
I think this should be a part of the `Interface`, not a trait extension of the `primitives_currency_swap::CurrencySwap` - because think of how you would use it, you'd have to make a newtype just to implement the trait at runtime, no?

Reply:
Gave this a more-in depth look - for sure we need to move this functionality into the `Interface`. The point of an `Interface` is that you implement it for the `Runtime`, and thus guarantee there are no conflicting implementations possible since it is a single-purpose type.

If we wanted to use this in runtime, we'd have to implement `pallet_currency_swap::benchmarking:: CurrencySwap` for `bridge_currency_swap::OneToOne`, at runtime crate, which are both foreign to that crate and it won't work in Rust.

Reply:
https://github.com/humanode-network/humanode/pull/679/commits/ab585b22dbebdb23315ea3de80ab2f0fe90666ef